tests/unit/test_linters.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+"""Unit tests for linters."""
+
+import pathlib
+import sys
+
+import pytest
+
+from charmcraft import linters
+from charmcraft.models.lint import LintResult
+
+
+def test_pip_check_not_venv(fake_path: pathlib.Path):
+    lint = linters.PipCheck()
+    assert lint.run(fake_path) == LintResult.NONAPPLICABLE
+    assert lint.text == "Charm does not contain a Python venv."
+
+
+def test_pip_invalid_venv(fake_path: pathlib.Path):
+    (fake_path / "venv").mkdir()
+    lint = linters.PipCheck()
+    assert lint.run(fake_path) == LintResult.NONAPPLICABLE
+    assert lint.text == "Python venv is not valid."
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Windows not [yet] supported.")
```

Comment:
Is this because a fixture or preliminary step fails?  Or should there be a test that mocks `os.platform` to exercise these lines?

```python
        if sys.platform == "win32":
            self.text = "Linter does not work on Windows."
            return self.Result.NONAPPLICABLE
```

Reply:
I think this might actually be removable. This was a skip I originally put in when I misdiagnosed an issue with the test passing on Windows, but the problem ended up being something else. Considering #1810 though, is this something I should fix?

Reply:
Yeah, I guess it doesn't really matter if it's tested or not since windows is on the chopping block